pkg/cli/create/routeedge.go

```diff
@@ -104,6 +104,12 @@ func (o *CreateEdgeRouteOptions) Run() error {
 		return err
 	}
 
+	// If the namespace is not the default, add it. This makes sure it will eventually
+	// appear in the manifest if the dry-run is enabled.
+	if o.CreateRouteSubcommandOptions.Namespace != corev1.NamespaceDefault {
```

Comment:
This is not correct, you should be setting the namespace only when explicitly requested, iow. you should check `o.CreateRouteSubcommandOptions.EnforceNamespace` and set it only then. I'd expect this to happen inside `UnsecuredRoute` method, though. See https://github.com/kubernetes/kubernetes/blob/80b602299ff0b8ae13596ad6d4634aa89cd99697/staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go#L248-L250 as a good example.

Reply:
@soltysh I just updated it, it should look better now. Thanks for the comment.